client/verta/tests/unit_tests/test_deployed_model.py

```diff
@@ -379,3 +382,134 @@ def test_predict_400_error_message_missing(mocked_responses) -> None:
+    """ Call batch_predict with a single batch. """
+    expected_df = pd.DataFrame({"A": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], "B": [11, 12, 13, 14, 15, 16, 17, 18, 19, 20]})
+    expected_df_json = expected_df.to_dict(orient="index")
+    mocked_responses.post(
+        BATCH_PREDICTION_URL,
+        json=expected_df_json,
+        status=200,
+        headers={"verta-request-id": "hereISaTESTidFROMtheUSER"},
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    # the input below is entirely irrelevant since it"s smaller than the batch size
+    prediction_df = dm.batch_predict(pd.DataFrame({"hi": "bye"}, index=[1]), 10)
+    # Since no index was provided, we can"t guarantee the index type for assertions
+    pd.testing.assert_frame_equal(expected_df.reset_index(drop=True), prediction_df.reset_index(drop=True))
+
+
+def test_batch_predict_with_one_batch_with_output_index(mocked_responses) -> None:
+    """ Call batch_predict with a single batch, where the output has an index. """
+    expected_df = pd.DataFrame({"A": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], "B": [11, 12, 13, 14, 15, 16, 17, 18, 19, 20]}, index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"])
+    expected_df_json = expected_df.to_dict(orient="index")
+    mocked_responses.post(
+        BATCH_PREDICTION_URL,
+        json=expected_df_json,
+        status=200,
+        headers={"verta-request-id": "hereISaTESTidFROMtheUSER"},
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    # the input below is entirely irrelevant since it"s smaller than the batch size
+    prediction_df = dm.batch_predict(pd.DataFrame({"hi": "bye"}, index=[1]), 10)
+    # Since an index WAS provided, we should be able to assert with indexes included
+    pd.testing.assert_frame_equal(expected_df, prediction_df)
+
+
+def test_batch_predict_with_five_batches_of_one_with_no_indexes(mocked_responses) -> None:
+    """ Call batch_predict with five batches. """
+    expected_d_list = [pd.DataFrame({"A": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]}),
+                       pd.DataFrame({"B": [11, 12, 13, 14, 15, 16, 17, 18, 19, 20]}),
+                       pd.DataFrame({"C": [21, 22, 23, 24, 25, 26, 27, 28, 29, 30]}),
+                       pd.DataFrame({"D": [31, 32, 33, 34, 35, 36, 37, 38, 39, 40]}),
+                       pd.DataFrame({"E": [41, 42, 43, 44, 45, 46, 47, 48, 49, 50]}),
+                       ]
+    for expected_d in expected_d_list:
+        mocked_responses.add(
+            responses.POST,
+            BATCH_PREDICTION_URL,
+            json=expected_d.to_dict(orient="index"),
+            status=200,
+            headers={"verta-request-id": "hereISaTESTidFROMtheUSER"},
```

Comment:
I think we should remove the header from these tests because it's unrelated haha (unless we do need it??)